src/pushcollector/_impl/dummy.py

```diff
@@ -1,4 +1,10 @@
 class DummyCollector(object):
+    def __enter__(self):
```

Comment:
There's a bit of a problem being shown up here.

The earlier situation for the Collector API was:
- it's not strictly required to subclass Collector in order to provide a backend (as shown here)
- existing backends might not provide enter/exit (as shown here)

This means if you now want to make `with Collector.get()` possible, you have a problem:
1. either you declare "all backends must implement enter/exit" which is a non-backwards-compatible change
2. or you declare `with Collector.get()` can only be used if you know the backend in use implemented enter/exit, which makes the API harder to use
3. or you make pushcollector somehow cope with the possibility that backends might not have enter/exit, and just make it do a no-op in that case

If you're adding enter/exit to existing classes, it seems like you're thinking of (1), but I think we should actually leave those classes alone and do (3).

It is unfortunate that I didn't think to make this use context manager protocol from the beginning, but at least by using CollectorProxy it should be possible to implement (3) easily at that layer (just check if self._delegate has enter/exit, and invoke them if and only if it does).

Reply:
You are right I was thinking (1), but now I think I agree that (3) makes more sense. It is easy enough to do something in CollectorProxy like:
```
def __enter__(self):
    if hasattr(self._delegate, "__enter__"):
        self._delegate = self._delegate.__enter__()
    return self
```
It would nice if the standard library had a function like `callable` that returns `True` if an object supports the context manager protocol, but I don't see anything like that even in `contextlib`.

Reply:
I'd rather not require a particular subclass.  I think duck typing is considered more idiomatic python, where the object should be accepted if it has the right interface and you don't artificially restrict the types that you accept.

If this started to mandate a certain type, for example doing an isinstance check and raising, it would be backwards-incompatible and I guess would break some useful patterns like using a MagicMock object as a collector during a test, since that wouldn't be an instance of Collector.